apps/web/app/api/v1/lib/api-key.ts

```diff
@@ -0,0 +1,53 @@
+import { apiKeyCache } from "@/lib/cache/api-key";
+import { Prisma } from "@prisma/client";
+import { ApiKey } from "@prisma/client";
+import { cache as reactCache } from "react";
+import { prisma } from "@formbricks/database";
+import { cache } from "@formbricks/lib/cache";
+import { getHash } from "@formbricks/lib/crypto";
+import { validateInputs } from "@formbricks/lib/utils/validate";
+import { ZString } from "@formbricks/types/common";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/types/errors";
+import { InvalidInputError } from "@formbricks/types/errors";
+
+export const getEnvironmentIdFromApiKey = reactCache(
+  async (apiKey: string): Promise<Pick<ApiKey, "environmentId"> | null> => {
```

Comment:
I think we can just return a `string` from here or `{environmentId: string}` 😅 The pick seems a bit unnecessary here